### PR TITLE
refactor(entrypoint): RHICOMPL-2858 rewrite cgroup-limits to bash

### PR DESCRIPTION
**Pull request by skateman**
Original implementation is [here](https://github.com/sclorg/container-common-scripts/blob/master/shared-scripts/core/usr/bin/cgroup-limits), this is intended for use in the ubi8-minimal container.

## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [x] Input Validation
- [x] Output Encoding
- [x] Authentication and Password Management
- [x] Session Management
- [x] Access Control
- [x] Cryptographic Practices
- [x] Error Handling and Logging
- [x] Data Protection
- [x] Communication Security
- [x] System Configuration
- [x] Database Security
- [x] File Management
- [x] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: scripts/set_cgroup_limits.sh [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+# Script for parsing cgroup information
+# Inspired by https://github.com/sclorg/container-common-scripts/blob/master/shared-scripts/core/usr/bin/cgroup-limits
+
+export MAX_MEMORY_LIMIT_IN_BYTES=9223372036854775807
+echo "MAX_MEMORY_LIMIT_IN_BYTES=${MAX_MEMORY_LIMIT_IN_BYTES}"
+
+# Determine memory limit from cgroups
+MEMORY_LIMIT_CGROUP="/sys/fs/cgroup/memory/memory.limit_in_bytes"
+if test -f "${MEMORY_LIMIT_CGROUP}"; then
```
Comment (vkrizan) [inline scripts/set_cgroup_limits.sh:11]: Please use `[[`

File: scripts/set_cgroup_limits.sh [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+if test -f "${CORES_CGROUP}"; then
+  CORES=$(cat $CORES_CGROUP)
+else
+  CORES=$(cat "/sys/fs/cgroup/cpuset.cpus.effective")
+fi
+
+CORE_COUNT=0
+
+# The group information about CPU count in the form of "A-B,C-D,..." that should be parsed into "B-A+1 + D-C+1 + ..."
+for GROUP in $(echo "${CORES}" | tr "," "\n"); do
```
Comment (vkrizan) [inline scripts/set_cgroup_limits.sh:43]: Nit: `<<< "${CORES}"` can be used instead of `echo`.

File: scripts/set_cgroup_limits.sh [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  if [[ "$GROUP" =~ (^[^-]+-[^-]+$) ]]; then
+    CORE_COUNT=$((CORE_COUNT + $(echo "${GROUP}" | awk '{ split($0, cores, "-"); print cores[2] - cores[1] + 1 }')))
```
Comment (vkrizan) [inline scripts/set_cgroup_limits.sh:45]: Alternative without awk
```
left=$(cut -d- -f1 <<< "${GROUP}")
right=$(cut -d- -f2 <<< "${GROUP}")
if [[ -n "$right" ]]; then
  CORE_COUNT=$((CORE_COUNT + $right - $left + 1))
else
  CORE_COUNT=$((CORE_COUNT + 1))
fi
```

(needs testing)

File: scripts/set_cgroup_limits.sh (hunk 2 above)
Comment (skateman) [reply scripts/set_cgroup_limits.sh:43]: I'm skipping this one, mostly because it's hard to understand (for me).

**Review by vkrizan — APPROVED** [review]
Noice. Feel free to add additional warning messages as needed.

File: scripts/set_cgroup_limits.sh (hunk 2 above)
Comment (Victoremepunto) [reply scripts/set_cgroup_limits.sh:43]: from 

> A variant of here documents, the format is:
>
>[n]<<< word
>The word undergoes tilde expansion, parameter and variable expansion, command substitution, arithmetic expansion, and quote removal. Filename expansion and word splitting are not performed. The result is supplied as a single string, with a newline appended, to the command on its standard input (or file descriptor n if n is specified).

**Pull request merged**